caddyhttp/proxy/proxy_test.go

```diff
@@ -1433,21 +1478,36 @@ func newWebSocketTestProxy(backendAddr string, insecure bool) *Proxy {
 			name:     backendAddr,
 			without:  "",
 			insecure: insecure,
+			timeout:  30 * time.Second,
 		}},
 	}
 }
 
 func newPrefixedWebSocketTestProxy(backendAddr string, prefix string) *Proxy {
 	return &Proxy{
 		Next:      httpserver.EmptyNext, // prevents panic in some cases when test fails
-		Upstreams: []Upstream{&fakeWsUpstream{name: backendAddr, without: prefix}},
+		Upstreams: []Upstream{&fakeWsUpstream{name: backendAddr, without: prefix, timeout: 30 * time.Second}},
+	}
+}
+
+// newWebSocketTestProxyWithTimeout returns a WebSocket proxy with a configurable upstream timeout
+func newWebSocketTestProxyWithTimeout(backendAddr string, insecure bool, timeout time.Duration) *Proxy {
```

Comment:
The same applies here. This is duplicate of `newPrefixedWebSocketTestProxy` except for the timeout param. It's better if you supply a timeout parameter to`newPrefixedWebSocketTestProxy` and use that value or the default.